dso/model.py

```diff
@@ -114,64 +110,25 @@ class Metadata:
     type: str
     relation: Relation | None = None
     creation_date: datetime.datetime | str = None
+    last_update: datetime.datetime | str = None
 
     def __post_init__(self):
         # only determine creation date on/post init because otherwise the creation date will
         # be the date when this module was imported
+        now = datetime.datetime.now()
         if not self.creation_date:
             # since the dataclass is frozen, `__setattr__` must be used
             # [ref](https://docs.python.org/3/library/dataclasses.html#frozen-instances)
-            object.__setattr__(self, 'creation_date', datetime.datetime.now())
-
-
-@dataclasses.dataclass(frozen=True)
-class CVE:
-    cve: str | None
-    cvss3Score: float
-    cvss: dso.cvss.CVSSV3 | dict | None
-    affected_package_name: str | None
-    affected_package_version: str | None
-    reportUrl: str
-    product_id: int
-    group_id: int
-    base_url: str
-    bdba_cfg_name: str
+            object.__setattr__(self, 'creation_date', now)
+        if not self.last_update:
+            object.__setattr__(self, 'last_update', now)
```

Comment:
some thoughts w.r.t. creation-date + mutability.

generally speaking, if we need our model objects to be mutable for good reasons, and there are no strong reasons for making them immutable, we might just as well consider making our model-classes mutable, instead of making them mutable and then patch them anyway :-)

In addition, I am not exactly sure whether implicitly setting creation-date + last-update to "now" is such a good idea.

If we want to enforce _all_ instances must _always_ have those attributes set, let is instead add a validation (raise ValueError or the like), so programming errors will be detected early. If we want to ensure instances must have them set in certain cases (e.g. prior to uploading to delivery-service), we might add validation logic there.

Reply:
In fact, we do not require this model object to be mutable since the `creation_date` and `last_update` can be set during initialisation already. However, this hack, to set them in the `__post_init__` function, was just for convenience.

I suggest I remove it from here again and update usages to explicitly set those attributes.